### PR TITLE
check errors in channels/api4/channel.go

### PR DESCRIPTION
**Pull request by pradeepmurugesan**
#### Summary
Error checks for channels/api4/channel.go

#### Ticket Link
Fixes https://github.com/mattermost/mattermost/issues/28352

#### Release Note

```release-note

```

### CODE REVIEW

**Review by hanzei — APPROVED** [review]
LGTM :+1:

**Comment by hanzei** [comment]
@rahimrahman Gentle reminder to review the PR :wink:

**Review by rahimrahman — APPROVED** [review]
I am not really up to snub with go, but what I see in here is replacing a few Write with some sort of error handling and logging.  As somebody who needs to learn, if @hanzei thinks this is good, then I defer to his expertise, and take this as learning experience.

Q: would `err` lets you know what method is causing the error?  If there's any benefit of tracking that, then that might be useful.
Q: can this be a util function instead of rewriting the same thing over and over?

File: server/.golangci.yml [hunk 1]
```diff
@@ -62,7 +62,6 @@ issues:
         channels/api4/apitestlib.go|\
         channels/api4/bot_test.go|\
         channels/api4/brand.go|\
-        channels/api4/channel.go|\
```
Comment (rahimrahman) [inline server/.golangci.yml:65]: Curiosity question, what's the reasoning of removing this?

Comment (hanzei) [reply server/.golangci.yml:64]: That list contains all exceptions for the `errcheck` linter. This PR fixed the `errcheck` issues in the `channels/api4/channel.go` file and hence, we can remove the exception.

**Comment by hanzei** [comment]
> I am not really up to snub with go, but what I see in here is replacing a few Write with some sort of error handling and logging. As somebody who needs to learn, if @hanzei thinks this is good, then I defer to his expertise, and take this as learning experience.

Sound good! Please hit me with any other questions you have!

> Q: would `err` lets you know what method is causing the error? If there's any benefit of tracking that, then that might be useful. 

It depends :wink: . Plain errors in Go don't contain stack information. However, there are libraries that add that. https://github.com/pkg/errors is a commonly used one. You can also add contextual information to errors. Please check out https://go.dev/blog/go1.13-errors for more details.

> Q: can this be a util function instead of rewriting the same thing over and over?

It could be. What is repetitive in the current code is the log message. I see your point of moving that to a utils function. However, I slightly favour the consistency in the code base. Every API method does error checks for writes in this same way. If we want to move the code into a helper, we can do that later.

**Pull request merged**